src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/CertificateExtensionsCommon.cs

```diff
@@ -22,15 +22,30 @@ internal static class CertificateExtensionsCommon
             string oidValue = GetExpectedOidValue<T>();
             PublicKey publicKey = certificate.PublicKey;
             Oid algorithmOid = publicKey.Oid;
+
             if (oidValue != algorithmOid.Value)
                 return null;
```

Comment:
Is this check still appropriate for ECDH?  Can't it use id-ecDH or id-ecPublicKey?

Reply:
(From a comment later, apparently not...)